inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -24,16 +26,84 @@ def test_parse_add_arguments_successfully(self):
                                      'deb-src http://example.com/ focal-security main'])
         self.assertEqual(f.filename, 'intel-gpu-jammy.list')
 
-    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
-    def test_create_add_manifest_successfully(self, m_connect):
+    def test_parse_add_arguments_deb822_format_separate_lines_successfully(self):
+        f = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--sources', 'X-Repolib-Name: Google Chrome',
+             'Enabled: yes',
+             'Types: deb',
+             'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+             'Suites: stable',
+             'Components: main',
+             '--filename', 'intel-gpu-jammy.list'])
+        self.assertEqual(f.gpgKeyUri, None)
+        self.assertEqual(f.gpgKeyName, None)
+        self.assertEqual(f.sources, ['X-Repolib-Name: Google Chrome',
+                                     'Enabled: yes',
+                                     'Types: deb',
+                                     'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+                                     'Suites: stable',
+                                     'Components: main'])
+        self.assertEqual(f.filename, 'intel-gpu-jammy.list')
+
+    def test_raise_application_add_with_only_one_gpgKeyUri_param(self):
+        with patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect') as m_connect:
+            p = self.arg_parser.parse_args(
+                ['source', 'application', 'add',
+                 '--gpgKeyUri', 'https://repositories.intel.com/gpu/intel-graphics.key',
+                 '--sources', 'deb http://example.com/ focal main restricted universe',
+                 'deb-src http://example.com/ focal-security main',
+                 '--filename', 'intel-gpu-jammy.list'])
+            with pytest.raises(InbcException,
+                               match="Source requires either both gpgKeyUri and gpgKeyName "
+                                     "to be provided, or neither of them."):
```

Comment:
**Best Practice**
**Suggestion:** Avoid using a broad exception match in tests, as it may hide other exceptions that are not intended to be caught. Specify the exact exception you expect.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            with pytest.raises(InbcException) as exc_info:
                Inbc(p, 'source', False)
            assert "Source requires either both gpgKeyUri and gpgKeyName to be provided, or neither of them." in str(exc_info.value)
```